clang-tools-extra/clang-doc/HTMLMustacheGenerator.cpp

```diff
@@ -0,0 +1,523 @@
+  llvm::SmallString<64> Path = I.getRelativeFilePath(CurrentDirectory);
+  llvm::sys::path::append(Path, I.getFileBaseName() + ".html");
+  llvm::sys::path::native(Path, llvm::sys::path::Style::posix);
+  Object Obj = Object();
+  Obj.insert({"Link", Path});
+  Obj.insert({"Name", I.Name});
+  Obj.insert({"QualName", I.QualName});
+  Obj.insert({"ID", llvm::toHex(llvm::toStringRef(I.USR))});
+  return Obj;
+}
+
+
+Value extractValue(const TypedefInfo &I) {
+  // Not Supported
+  return nullptr;
+}
+
+Value extractValue(const CommentInfo &I) {
+  Object Obj = Object();
+  Value Child = Object();
+  
+  if (I.Kind == "FullComment") {
+    Value ChildArr = Array();
+    for (const auto& C: I.Children)
+      ChildArr.getAsArray()->emplace_back(extractValue(*C));
+    Child.getAsObject()->insert({"Children", ChildArr});
+    Obj.insert({"FullComment", Child});
+  }
+  if (I.Kind == "ParagraphComment") {
+    Value ChildArr = Array();
+    for (const auto& C: I.Children)
+      ChildArr.getAsArray()->emplace_back(extractValue(*C));
+    Child.getAsObject()->insert({"Children", ChildArr});
+    Obj.insert({"ParagraphComment", Child});
+  }
+  if (I.Kind == "BlockCommandComment") {
+    Child.getAsObject()->insert({"Command", I.Name});
+    Value ChildArr = Array();
+    for (const auto& C: I.Children)
+      ChildArr.getAsArray()->emplace_back(extractValue(*C));
+    Child.getAsObject()->insert({"Children", ChildArr});
+    Obj.insert({"BlockCommandComment", Child});
+  }
+  if (I.Kind == "TextComment")
+    Obj.insert({"TextComment", I.Text});
+  
+  return Obj;
+}
+
+Value extractValue(const FunctionInfo &I, StringRef ParentInfoDir,
+                   const ClangDocContext &CDCtx) {
+  Object Obj = Object();
+  Obj.insert({"Name", I.Name});
+  Obj.insert({"ID", llvm::toHex(llvm::toStringRef(I.USR))});
+  Obj.insert({"Access", getAccessSpelling(I.Access).str()});
+  Obj.insert({"ReturnType", extractValue(I.ReturnType.Type, ParentInfoDir)});
+  
+  Value ParamArr = Array();
+  for (const auto Val : llvm::enumerate(I.Params)) {
+    Value V = Object();
+    V.getAsObject()->insert({"Name", Val.value().Name});
+    V.getAsObject()->insert({"Type", Val.value().Type.Name});
+    V.getAsObject()->insert({"End",  Val.index() + 1 == I.Params.size()});
+    ParamArr.getAsArray()->emplace_back(V);
+  }
+  Obj.insert({"Params", ParamArr});
+  
+  if (!I.Description.empty()) {
+    Value ArrDesc = Array();
+    for (const CommentInfo& Child : I.Description) 
+      ArrDesc.getAsArray()->emplace_back(extractValue(Child));
+    Obj.insert({"FunctionComments", ArrDesc});
+  }
+  if (I.DefLoc.has_value()) {
+    Location L = *I.DefLoc;
+    if (CDCtx.RepositoryUrl.has_value())
+      Obj.insert({"Location", extractValue(L,
+                                           StringRef{*CDCtx.RepositoryUrl})});
+    else
+      Obj.insert({"Location", extractValue(L)});  
+  }
+  return Obj;
+}
+
+Value extractValue(const EnumInfo &I, const ClangDocContext &CDCtx) {
```

Comment:
If this is an `llvm::json::Value` then lets keep at least the `json::` prefix. Value is a very overloaded term in LLVM.